analyzer/src/traversal/assignments.rs

```diff
@@ -39,6 +66,30 @@ pub fn assign(
     unreachable!()
 }
 
+pub fn check_assign_target(
```

Comment:
I now notice that I didn't apply the same check to aug_assign. That can wait for a future pr.